multimodal/src/autogluon/multimodal/learners/abstract_mm_learner.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import warnings
+
+from abc import ABC, abstractmethod, abstractclassmethod, abstractstaticmethod, abstractproperty
+from typing import Optional, Union, List
+
+import pandas as pd
+
+from autogluon.multimodal.problem_types import ProblemTypeProperty
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractMMLearner(ABC):
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        pass
+
+    @abstractproperty
+    def path(self) -> Optional[str]:
+        pass
+
+    @abstractproperty
+    def label(self) -> Optional[str]:
+        pass
+
+    @abstractproperty
+    def problem_type(self) -> Optional[str]:
+        pass
+
+    @abstractproperty
+    def problem_property(self) -> Optional[ProblemTypeProperty]:
+        pass
+
+    @abstractproperty
+    def class_labels(self) -> Optional[List[str]]:
+        pass
+
+    @abstractproperty
+    def positive_class(self) -> Optional[str]:
+        pass
+
+    @abstractmethod
+    def fit(
+        self,
+        train_df: pd.DataFrame,
+        val_df: pd.DataFrame,
+        validation_metric_name: str,
+        minmax_mode: str,  # this is determined solely on validation_metric_name
+        save_path: str,
+        ckpt_path: str,  # these two are synced
```

Comment:
Why is `save_path` a fit arg instead of an init_arg?

Reply:
Similar to `validation_metric_name`, this is to support calling `.fit()` multiple times with different sets of `hyperparameters` so that users may save a model for each set of `hyperparameters`.